addons/wowl/static/src/core/utils.ts

```diff
@@ -11,3 +11,23 @@ export function makeContext(...contexts: ContextDescription[]): Context {
   }
   return context;
 }
+
+/**
+ * Returns a string formatted using given values.
+ * If the value is an object, its keys will replace `%(key)s` expressions.
+ * If the values are a set of strings, they will replace `%s` expressions.
+ * If no value is given, the string will not be formatted.
+ */
+export function sprintf(s: string, ...values: string[]): string {
+  if (values.length === 1 && typeof values[0] === "object") {
+    const valuesDict = values[0] as { [key: string]: string };
+    for (const value in valuesDict) {
+      s = s.replace(`%(${value})s`, valuesDict[value]);
```

Comment:
there should be a way to do it in one replace operation. something like this: (not tested!!!)
```js
s = s.replace(/\%\(*\)s/g, value => valuesDict[value);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace